### PR TITLE
Unify trace and metric stdout exporters

### LINKED ISSUE #955: Unify metric and trace the STDOUT exporters
Similar to the OTLP exporter, we should unify the STDOUT exporters that exist in the trace and metric directories.

### PR DESCRIPTION
**Pull request by MrAlias**
- Unify both exporters into a single `Exporter`.
- Create a unified `Config` that conforms to project patterns.
- Align output formats.
   - Export spans as lists.
   - Remove outer object wrapper on metric, and export as a list of metrics.
- Add an overview README in the `exporters` directory.

Resolves #955

### CODE REVIEW

File: example/basic/main.go [hunk 1]
```diff
@@ -37,37 +33,15 @@ var (
...
 	if err != nil {
-		log.Panicf("failed to initialize metric stdout exporter %v", err)
+		log.Fatalf("failed to initialize stdout export pipeline: %v", err)
 	}
-	return pusher
-}
-
-func main() {
-	defer initMeter().Stop()
-	initTracer()
```
Comment (MrAlias) [inline example/basic/main.go:70]: This is a great example of what this unification is saving the user from. :smile:

File: exporters/stdout/example_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporters/stdout"
+)
+
+const (
+	instrumentationName    = "github.com/instrumentron"
```
Comment (lizthegrey) [inline exporters/stdout/example_test.go:29]: typo

File: exporters/stdout/exporter.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	_ trace.SpanBatcher = &Exporter{}
+)
+
+// NewExporter creates an Exporter with the passed options.
+func NewExporter(options ...Option) (*Exporter, error) {
+	config, err := Configure(options...)
+	if err != nil {
+		return nil, err
+	}
+	return &Exporter{
```
Comment (lizthegrey) [inline exporters/stdout/exporter.go:44]: should there not be the option to disable the trace or metric exporter e.g. to send all metrics to stdout but not to send traces?

Comment (jmacd) [reply exporters/stdout/exporter.go:44]: I could imagine this, but I wonder if it's going to be used much. I can't imagine many uses for half of the stdout exporter. If you're setting up one and not the other, you could avoid calling InstallNewPipeline instead and call only one of the global.Set methods.

Comment (lizthegrey) [reply exporters/stdout/exporter.go:44]: I can imagine quite a few -- for one, in Honeycomb's case we might have a trace exporter we're fully confident in with no need to print values locally, but be testing our metric export pipeline and want to compare the values side by side with what stdout is returning. But we wouldn't want every trace to get printed to stdout.

or vice versa in the case of someone who is using metrics successfully but wants to test their trace export pipeline.

File: exporters/stdout/example_test.go (hunk 2 above)
Comment (MrAlias) [reply exporters/stdout/example_test.go:29]: Guessing you mean `instrumentron`? That was my hidden joke about some robot instrumentation package. I can change it if you feel strong, but it was intentional.

:smile:

File: exporters/stdout/exporter.go (hunk 3 above)
Comment (MrAlias) [reply exporters/stdout/exporter.go:44]: If you see a use @lizthegrey I'll add it here. I thought that kind of enablement/disablement was going to be handled as part of the Views API, but can add something here for the time being.

Comment (MrAlias) [reply exporters/stdout/exporter.go:44]: @lizthegrey PTAL

File: exporters/stdout/example_test.go (hunk 2 above)
Comment (lizthegrey) [reply exporters/stdout/example_test.go:29]: got it!

**Pull request merged**